micro-infra-spring-base/src/main/groovy/com/ofg/infrastructure/web/resttemplate/fluent/ServiceRestClientConfiguration.groovy

```diff
@@ -45,11 +42,31 @@ class ServiceRestClientConfiguration {
     @Value('${retry.threads:10}') int retryPoolThreads
 
     @Bean
-    ServiceRestClient serviceRestClient(ServiceResolver serviceResolver, ServiceConfigurationResolver configurationResolver) {
-        return new ServiceRestClient(microInfraSpringRestTemplate(), serviceResolver, configurationResolver)
+    ServiceRestClient serviceRestClient(
+            @Qualifier('micro-infra-spring') RestOperations restOperations,
+            ServiceResolver serviceResolver,
+            ServiceConfigurationResolver configurationResolver
+    ) {
+        return new ServiceRestClient(restOperations, serviceResolver, configurationResolver)
     }
 
+    /**
+     * For my solution to work all micro-infra beans should:
+     *    1. be candidates for autowiring
+     *    2. be injected by type + qualifier within micro-infra
+     *    3. have names prefixed with 'microInfra'
+     *    4. be defined with @ConditionalOnMissingBean
+     *
+     * The above can be achieved easily by creating an ast transformation that will add
+     * `@Qualifier('micro-infra')` to all method parameters (and - optionally - all @Bean definitions).
+     * The transformation can also report compile errors when the bean name does not start with 'microInfra'.
+     *
+     * Adding @ConditionalOnMissingBean can be done either by the transformation or by using a meta-annotation
+     * that will carry all of [@Bean, @Qualifier('micro-infra-spring'), @ConditionalOnMissingBean] declarations.
+     */
     @Bean
+    @Qualifier('micro-infra-spring')
```

Comment:
Dedicated annotation for type safety

    @Qualifier('micro-infra-spring')
    @Retention(RUNTIME)
    @Target({TYPE, METHOD, FIELD, PARAMETER})
    public @interface MicroInfraSpring {}